### PR TITLE
workaround old gcc std::regex_iterator issue

### PR DESCRIPTION
**Pull request by hjanetzek**
- and this version is nicer anyway :)

### CODE REVIEW

File: core/src/gl/shaderProgram.cpp [hunk 1]
```diff
@@ -284,13 +284,10 @@ std::string ShaderProgram::applySourceBlocks(const std::string& source, bool fra
 
     auto sourcePos = source.begin();
 
-    auto linesBegin = std::sregex_iterator(source.begin(), source.end(), pragmaLine);
-    auto linesEnd = std::sregex_iterator();
+    for (auto it = source.begin(); std::regex_search(it, source.end(), sm, pragmaLine);) {
+        it += sm.position() + sm.length();
```
Comment (tallytalwar) [inline core/src/gl/shaderProgram.cpp]: Did we not just check that the `smatch::position` is different in earlier versions of gcc? Is it safe to relly on it then?

Or maybe it gives different results when used with regex_iterators!

Comment (hjanetzek) [inline core/src/gl/shaderProgram.cpp]: yeah, only std::regex_iterator seems to be broken a rather obscure feature. if regex_search gave the wrong position someone would probably have noticed before release :)

Comment (tallytalwar) [inline core/src/gl/shaderProgram.cpp]: Plus, it does work on older gcc and newer (ofcourse).
Merging when travis goes green.
Thanks @hjanetzek

**Comment by blair1618** [comment]
Ah, this _is_ nicer! LGTM

**Pull request merged**